### PR TITLE
Initial check in for iam custom role template

### PR DESCRIPTION
**Pull request by sourced-vince**

### CODE REVIEW

File: community/faas/templates/organization_custom_role.py.schema [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    type: string
+    pattern: ^[a-zA-Z][0-9a-zA-Z]{7,63}$
+    description: A unique ID for the role.
+  title:
+    type: string
+    default: "\"\""
+    description: The title of the custom role.
+  description:
+    type: string
+    default: "\"\""
```
Comment (tiadobatima1) [inline community/faas/templates/organization_custom_role.py.schema]: I'd remove default values for `title` and `description`. If they're not provided, we shouldn't add them to the resource.

File: community/faas/templates/project_custom_role.py.schema [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+required:
+  - roleId
+  - includedPermissions
+
+properties:
+  roleId:
+    type: string
+    pattern: ^[a-zA-Z][0-9a-zA-Z]{7,63}$
+    description: A unique ID for the role.
+  title:
```
Comment (tiadobatima1) [inline community/faas/templates/project_custom_role.py.schema]: I'd remove default values for `title` and `description`. If they're not provided, we shouldn't add them to the resource.

File: community/faas/tests/integration/iam-custom-role.bats [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+@test "Creating deployment ${DEPLOYMENT_NAME} from ${CONFIG}" {
+  gcloud deployment-manager deployments create "${DEPLOYMENT_NAME}" --config "${CONFIG}" --project "${FAAS_PROJECT_ID}"
+}
+
+@test "Verifying project iam roles were created in deployment ${DEPLOYMENT_NAME}" {
+  run gcloud iam roles list --project "${FAAS_PROJECT_ID}" --filter="name:myCustomProjectRole${RAND}"
+  [ "$status" -eq 0 ]
+  [[ "${lines[1]}" = "description: My Project Role Description" ]]
```
Comment (tiadobatima1) [inline community/faas/tests/integration/iam-custom-role.bats]: Are we 100% sure the response from `gcloud iam roles list` will always return these fields in this exact order? If not, I'd use regex comparisons like:
```
[[ "$output" =~ "My Project Role Description" ]]
```

File: community/faas/tests/fixtures/configs/iam-custom-role.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Example on how to use the iam-custom-role template
+#
+# Variables:
+#   ORGANIZATION_ID: The organization ID of your organization.
```
Comment (tiadobatima1) [inline community/faas/tests/fixtures/configs/iam-custom-role.yaml]: FAAS_ORGANIZATION_ID

**Comment by sourced-vince** [comment]
Fixed.

**Pull request closed without merging**